package/MDAnalysis/analysis/base.py

```diff
@@ -430,25 +517,219 @@ def run(self, start=None, stop=None, step=None, frames=None,
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+
+        Raises
+        ------
+        ValueError
+            if *both* `frames` and at least one of `start`, `stop`, or `frames`
+            is provided (i.e., set to another value than ``None``)
+
+        Returns
+        -------
+        computation_groups : list of (n, 2) shaped np.ndarrays with frame indices and numbers
+        """
+        if frames is None:
+            start, stop, step = self._trajectory.check_slice_indices(start, stop, step)
+            used_frames = np.arange(start, stop, step)
+        elif not all(opt is None for opt in [start, stop, step]):
+            raise ValueError("start/stop/step cannot be combined with frames")
+        else:
+            used_frames = frames
+        
+        if all((isinstance(obj, bool) for obj in used_frames)):
+            arange = np.arange(len(used_frames))
+            used_frames = arange[used_frames]
+        
+        # this numpy thing is similar to list(enumerate(frames))
+        enumerated_frames = np.vstack([np.arange(len(used_frames)), used_frames]).T
+
+        return np.array_split(enumerated_frames, n_parts)
+
+
+    def _configure_client(self, backend: str, n_workers: int, client):
+        """
+        Configure parameters necessary for running a distributed workload.
+
+        Parameters
+        ----------
+        backend : str, optional 
+            scheduling type:
+              - 'dask' implies `dask` scheduler (need to install MDAnalysis[dask] for that to work)
+              - 'multiprocessing' works with standard python multiprocessing module
+              - None if you want to run things locally
+        n_workers : int, optional
+            number of workers (local or remote processes).
+        """
+        if backend is not None and backend not in self.available_backends:
+            raise ValueError(f"{backend=} is not in {self.available_backends=} for class {self.__class__.__name__}")
+        return ParallelExecutor(backend=backend, n_workers=n_workers, client=client)
+
+
+    def run(self, 
+            start: int = None, stop: int = None, step: int = None, 
+            frames: Iterable = None,
+            verbose: bool = None, 
+            n_workers: int = None, 
+            n_parts: int = None,
+            backend: str = 'local',
+            *, 
+            client: object = None,
+            progressbar_kwargs={},
+            ):
+        """Perform the calculation
+
+        Parameters
```

Comment:
Some parameters missing from the docstring? Please double check this docstring.

Reply:
Yep, you're right -- fixed.